### ISSUE TITLE
WasmFS: Support symlinks

### ISSUE THREAD

**Issue by tlively** [emscripten-core/emscripten#15948]
WasmFS should support symlinks as a third kind of file besides DataFile and Directory.

**Comment by curiousdannii** [comment]
Though it would be nice for custom backends to not need to support them. If it's like the old FS then the make link function could just fail?

**Comment by tlively** [comment]
Good point. It would be nice to make a default implementation of symlinks that is implemented in terms of normal DataFiles. Backends that want to do something else for symlinks could override that default with their specialized implementation.

**Comment by curiousdannii** [comment]
My app shouldn't ever need symlinks. Any sort of default implementation like that would surely use more code than having the `symlink` syscall simply fail, right?

**Comment by tlively** [comment]
Oh yes, a custom backend could certainly choose not to implement symlinks at all.

**Comment by tlively** [comment]
We now have `Symlink` class after #16074, but we still need to follow links when parsing paths.
